src/agent/src/util.rs

```diff
@@ -104,7 +108,9 @@ mod tests {
 
             let mut vec_locked = vec_ref.lock();
 
-            let mut v = vec_locked.as_deref_mut().unwrap();
+            let mut v = vec_locked
+                .as_deref_mut()
+                .map_err(|e| std::io::Error::new(std::io::ErrorKind::Other, e.to_string()))?;
```

Comment:
Since this is only used in the unit tests, `unwrap()` is fine - it's just a specialized `assert`.